### ISSUE TITLE
generateChangelog with CockroachDB: addForeignKeyConstraint must have correct column order

### ISSUE THREAD

**Issue by SteveDonie** [liquibase/liquibase#1022]
**Description**
This is fairly easy to reproduce using sample code provided by the folks at CoackroachDB. The issue is that when using the generateChangeLog command against a table that has primary keys, the generated changelog has a changeset to create the table, followed by a second changeset to add the primary key to the table. This changelog is not able to be deployed to a new clean database without errors.

**To Reproduce**
Follow the instruction in the readme file at https://github.com/timveil-cockroach/liquibase-example

The steps under "Test 2" show the failing condition. Note that this bug is somewhat "masked" in the example by the existence of #1021 

The issue with the `<addForeignKeyConstraint>` changes generated is that the order of column names in the `referencedColumnNames` attribute needs to match the actual order of the columns in the table.

For example, in the current code, a 'bad' changeset looks like this:
```
    <changeSet author="steve (generated)" id="1582906076997-23">
        <addForeignKeyConstraint baseColumnNames="c_w_id,c_d_id" baseTableName="customer" constraintName="fk_c_w_id_ref_district" deferrable="false" initiallyDeferred="false" onDelete="NO ACTION" onUpdate="NO ACTION" referencedColumnNames="d_w_id,d_id" referencedTableName="district" validate="true"/>
    </changeSet>
```
Note that the order of the columns in `referencedColumnNames` is `d_w_id,d_id`. If we look at the generated `<createTable>` statement for the referenced table, we see that the actual order of the columns is `d_id,d_w_id` as seen below:
```
    <changeSet author="steve (generated)" id="1582906076997-2">
        <createTable tableName="district">
            <column name="d_id" type="BIGINT">
                <constraints nullable="false"/>
            </column>
            <column name="d_w_id" type="BIGINT">
                <constraints nullable="false"/>
            </column>
            <column name="d_name" type="VARCHAR(10)"/>
            <column name="d_street_1" type="VARCHAR(20)"/>
            <column name="d_street_2" type="VARCHAR(20)"/>
            <column name="d_city" type="VARCHAR(20)"/>
            <column name="d_state" type="CHAR(2)"/>
            <column name="d_zip" type="CHAR(9)"/>
            <column name="d_tax" type="numeric(4, 4)"/>
            <column name="d_ytd" type="numeric(12, 2)"/>
            <column name="d_next_o_id" type="BIGINT"/>
        </createTable>
    </changeSet>
```

**Expected behavior**
The generated changeset should look like this:
```
    <changeSet author="steve (generated)" id="1582906076997-23">
        <addForeignKeyConstraint baseColumnNames="c_w_id,c_d_id" baseTableName="customer" constraintName="fk_c_w_id_ref_district" deferrable="false" initiallyDeferred="false" onDelete="NO ACTION" onUpdate="NO ACTION" referencedColumnNames="d_id,d_w_id" referencedTableName="district" validate="true"/>
    </changeSet>
```

**Additional context**
The repro steps require the ability to run a docker container and maven. I used maven 3.6.

┆Issue is synchronized with this [Jira Bug](https://datical.atlassian.net/browse/LB-135) by [Unito](https://www.unito.io/learn-more)
┆Fix Versions: Liquibase 3.10

**Comment by molivasdat** [comment]
@SteveDonie Thanks for writing this issue up.  We are actively working on this issue and should be resolved soon.

**Comment by nvoxland** [comment]
This issue has been verified to be fixed in 3.8.9
